ir/ir.def

```diff
@@ -289,6 +291,7 @@ class P4Table : Declaration, IAnnotated, IApply {
 
     Annotations getAnnotations() const override { return annotations; }
     Type_Method getApplyMethodType() const override;
+    ParameterList getApplyParameters() const override { return nullptr; }
```

Comment:
I think this should be an empty parameter list.
You don't want to check for nullptr everywhere.
Moreover, this will make it easier to add parameters in the future if we decide to do that.